deployer/src/infra/csi.rs

```diff
@@ -0,0 +1,77 @@
+use super::*;
+use tokio::{
+    net::UnixStream,
+    time::{sleep, Duration},
+};
+use tonic::transport::{Endpoint, Uri};
+use tower::service_fn;
+
+use rpc::csi::{identity_client::IdentityClient, GetPluginInfoRequest};
+
+const CSI_SOCKET: &str = "/var/tmp/csi.sock";
+
+#[async_trait]
+impl ComponentAction for Csi {
+    fn configure(&self, options: &StartOptions, cfg: Builder) -> Result<Builder, Error> {
+        Ok(if options.no_csi {
+            cfg
+        } else {
+            if options.build {
+                std::process::Command::new("cargo")
+                    .args(&["build", "-p", "rest", "--bin", "rest"])
+                    .status()?;
+            }
+
+            let binary = Binary::from_dbg("csi-controller")
+                .with_args(vec!["--rest-endpoint", "http://rest:8081"])
+                // Make sure that CSI socket is always under shared directory
+                // regardless of what its default value is.
+                .with_args(vec!["--csi-socket", CSI_SOCKET]);
+
+            cfg.add_container_spec(
+                ContainerSpec::from_binary("csi-controller", binary)
+                    .with_bypass_default_mounts(true)
+                    .with_bind("/var/tmp", "/var/tmp"),
+            )
+        })
+    }
+    async fn start(&self, options: &StartOptions, cfg: &ComposeTest) -> Result<(), Error> {
+        if !options.no_csi {
+            cfg.start("csi-controller").await?;
+        }
+        Ok(())
+    }
+
+    async fn wait_on(&self, options: &StartOptions, _cfg: &ComposeTest) -> Result<(), Error> {
+        if options.no_csi {
+            return Ok(());
+        }
+
+        // Step 1: Wait till CSI controller's gRPC server is registered and is ready
+        // to serve API requests.
+        let channel = loop {
+            match Endpoint::try_from("http://[::]:50051")?
+                .connect_with_connector(service_fn(|_: Uri| UnixStream::connect(CSI_SOCKET)))
```

Comment:
does the connect have a timeout? we might want to set the sleeps lower, to ms here and on the sleeps, just like in `wait_url_timeout` because the deployer is brought up and down many times during test and the setup time will accrue.

Reply:
OK, I will reduce the timeout to 100ms here.